File: sample-conference-java/app/src/main/java/com/quickblox/sample/videochat/conference/java/activities/SettingsActivity.java
package com.quickblox.sample.videochat.conference.java.activities;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.EditTextPreference;
import android.preference.Preference;

import com.quickblox.sample.videochat.conference.java.R;
import com.quickblox.sample.videochat.conference.java.fragments.SettingsFragment;
import com.quickblox.sample.videochat.conference.java.utils.Consts;
import com.quickblox.sample.videochat.conference.java.utils.ToastUtils;
import com.quickblox.sample.videochat.conference.java.views.SeekBarPreference;


public class SettingsActivity extends BaseActivity implements SharedPreferences.OnSharedPreferenceChangeListener {
    private static final int MAX_VIDEO_START_BITRATE = 2000;

    private String bitrateStringKey;
    private SettingsFragment settingsFragment;
    private boolean isVideoSettings;

    public static void start(Context context, boolean isVideoSettings) {
        Intent intent = new Intent(context, SettingsActivity.class);
        intent.putExtra(Consts.EXTRA_SETTINGS_TYPE, isVideoSettings);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        isVideoSettings = getIntent().getBooleanExtra(Consts.EXTRA_SETTINGS_TYPE, true);
        initActionBar();

        Bundle bundle = new Bundle();
        bundle.putBoolean(Consts.EXTRA_SETTINGS_TYPE, isVideoSettings);

        settingsFragment = new SettingsFragment();
        settingsFragment.setArguments(bundle);
        getFragmentManager().beginTransaction()
                .replace(android.R.id.content, settingsFragment)
                .commit();
        bitrateStringKey = getString(R.string.pref_startbitratevalue_key);
    }

    private void initActionBar() {
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setTitle(isVideoSettings ? R.string.menu_settings_video : R.string.menu_settings_audio);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        SharedPreferences sharedPreferences =
                settingsFragment.getPreferenceScreen().getSharedPreferences();
        sharedPreferences.registerOnSharedPreferenceChangeListener(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        SharedPreferences sharedPreferences =
                settingsFragment.getPreferenceScreen().getSharedPreferences();
        sharedPreferences.unregisterOnSharedPreferenceChangeListener(this);
    }

    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
        if (key.equals(bitrateStringKey)) {
            int bitrateValue = sharedPreferences.getInt(bitrateStringKey, Integer.parseInt(
                    getString(R.string.pref_startbitratevalue_default)));
            if (bitrateValue == 0) {
                setDefaultstartingBitrate(sharedPreferences);
                return;
            }
            if (bitrateValue > MAX_VIDEO_START_BITRATE) {
                ToastUtils.longToast(getApplicationContext(), "Max value is:" + MAX_VIDEO_START_BITRATE);
                setDefaultstartingBitrate(sharedPreferences);
            }
        }
    }

    private void setDefaultstartingBitrate(SharedPreferences sharedPreferences) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt(bitrateStringKey,
                Integer.parseInt(getString(R.string.pref_startbitratevalue_default)));
        editor.apply();
        updateSummary(sharedPreferences, bitrateStringKey);
    }

    private void updateSummary(SharedPreferences sharedPreferences, String key) {
        Preference updatedPref = settingsFragment.findPreference(key);
        // Set summary to be the user-description for the selected value
        if (updatedPref instanceof EditTextPreference) {
            ((EditTextPreference) updatedPref).setText(sharedPreferences.getString(key, ""));
        } else if (updatedPref instanceof SeekBarPreference) {
            updatedPref.setSummary(String.valueOf(sharedPreferences.getInt(key, 0)));
        } else {
            updatedPref.setSummary(sharedPreferences.getString(key, ""));
        }
    }
}

File: sample-videochat-conference-kotlin/src/main/java/com/quickblox/sample/videochat/conference/kotlin/activities/SettingsActivity.kt
package com.quickblox.sample.videochat.conference.kotlin.activities

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.EditTextPreference
import com.quickblox.sample.videochat.conference.kotlin.R
import com.quickblox.sample.videochat.conference.kotlin.fragments.SettingsFragment
import com.quickblox.sample.videochat.conference.kotlin.utils.longToast
import com.quickblox.sample.videochat.conference.kotlin.view.SeekBarPreference


class SettingsActivity : BaseActivity(), SharedPreferences.OnSharedPreferenceChangeListener {

    private lateinit var bitrateStringKey: String
    private lateinit var settingsFragment: SettingsFragment

    companion object {
        private const val MAX_VIDEO_START_BITRATE = 2000

        fun start(context: Context) {
            context.startActivity(Intent(context, SettingsActivity::class.java))
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.title = getString(R.string.actionbar_title_settings)

        // Display the fragment as the main content.
        settingsFragment = SettingsFragment()
        fragmentManager.beginTransaction()
                .replace(android.R.id.content, settingsFragment)
                .commit()
        bitrateStringKey = getString(R.string.pref_startbitratevalue_key)
    }

    override fun onResume() {
        super.onResume()
        val sharedPreferences = settingsFragment.preferenceScreen.sharedPreferences
        sharedPreferences.registerOnSharedPreferenceChangeListener(this)
    }

    override fun onPause() {
        super.onPause()
        val sharedPreferences = settingsFragment.preferenceScreen.sharedPreferences
        sharedPreferences.unregisterOnSharedPreferenceChangeListener(this)
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
        if (key == bitrateStringKey) {
            val bitrateValue = sharedPreferences?.getInt(bitrateStringKey, Integer.parseInt(
                    getString(R.string.pref_startbitratevalue_default)))
            if (bitrateValue == 0) {
                setDefaultStartingBitrate(sharedPreferences)
                return
            }
            if (bitrateValue as Int > MAX_VIDEO_START_BITRATE) {
                longToast("Max value is:$MAX_VIDEO_START_BITRATE")
                setDefaultStartingBitrate(sharedPreferences)
            }
        }
    }

    private fun setDefaultStartingBitrate(sharedPreferences: SharedPreferences) {
        val editor = sharedPreferences.edit()
        editor.putInt(bitrateStringKey, Integer.parseInt(getString(R.string.pref_startbitratevalue_default))).apply()
        updateSummary(sharedPreferences, bitrateStringKey)
    }

    private fun updateSummary(sharedPreferences: SharedPreferences, key: String) {
        val updatedPref = settingsFragment.findPreference(key)
        // Set summary to be the user-description for the selected value
        if (updatedPref is EditTextPreference) {
            updatedPref.text = sharedPreferences.getString(key, "")
        } else if (updatedPref is SeekBarPreference) {
            updatedPref.setSummary(sharedPreferences.getInt(key, 0).toString())
        } else {
            updatedPref?.summary = sharedPreferences.getString(key, "")
        }
    }
}